src/main/java/server/UserService.java

```diff
@@ -0,0 +1,21 @@
+package server;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+class UserService {
+    private final HashSet<String> users = new HashSet<String>();

```

Comment:
HashSets are much faster than lists. Checking for the existence of an element in a Set is O(1), but in a list it is O(N). What this means is that Sets will be able to tell you if an element exists within the set regardless of the number of elements in a set.